### ISSUE TITLE
[Bug] MiMo-V2.5: any request with audio hangs the whole server under --enable-dp-attention (audio encoder reduces over the wrong TP group)

### ISSUE THREAD

**Issue by yuweih205** [sgl-project/sglang#37059]
### Describe the bug

Serving **MiMo-V2.5** (`MiMoV2ForCausalLM`) with DP attention — the configuration the model effectively mandates on an 8-GPU node, since its fused `qkv_proj` requires effective attention TP = 4:

```
--tp 8 --dp 2 --enable-dp-attention --enable-dp-lm-head --mm-enable-dp-encoder --quantization fp8
```

any request **containing audio** permanently hangs the whole server. Text-only and video-without-audio requests work fine.

| case | result |
|---|---|
| text only | ✅ |
| video without audio | ✅ |
| **audio (any request carrying an audio item)** | ❌ hangs 600 s, then the server dies |
| everything after that | ❌ connection refused |

Server log at the moment of death:

```
[PG ID 4 PG GUID 23(attention_tp:device) Rank 3] Process group watchdog thread terminated with exception:
[Rank 3] Watchdog caught collective operation timeout: WorkNCCL(SeqNum=14708, OpType=_ALLGATHER_BASE,
NumelIn=38144, NumelOut=152576, Timeout(ms)=600000) ran for 600083 milliseconds before timing out.
```

followed by gloo `Connection closed by peer` on the sibling ranks. Reproduced twice independently (official FP8 weights and a bf16 export) on 8×H200; identical signature both times.

### Root cause

`AudioEncoderAttention` builds its `VisionAttention` without `use_data_parallel` / `use_dp_attention_reduce` (`python/sglang/srt/models/mimo_audio.py:503`). Consequently:

- The audio encoder's attention weights **are sharded over the attention-TP group** — `_remap_audio_tokenizer_state_dict` slices by `attn_tp_rank`/`attn_tp_size` (`mimo_audio.py:1122-1173`), and `VisionAttention` defaults to `tp_size = attn_tp_size` (`layers/attention/vision.py:1072`).
- But its output projection's `RowParallelLinear` reduces over the **full TP group**: with `use_dp_attention_reduce=False` the reduce path takes `tensor_model_parallel_all_reduce` (`layers/linear.py:1666`) — 8 ranks, not 4.
- The audio encoder is only entered by ranks whose local batch actually has audio items: `general_mm_embed_routine` skips modalities with zero items (`managers/mm_utils.py:420`, `if len(items) != 0:`).

So when one DP group's batch carries audio and the other's does not, the audio-bearing group issues `encoder_layers × all_reduce` on the full-TP communicator that the other group never joins. The TP communicator desynchronizes and every subsequent collective piles up behind it.

The watchdog blames `attention_tp` only because that is the sole c10d-instrumented communicator in this setup: the full TP group runs through custom-allreduce/pynccl (no watchdog), while `attention_tp` falls through to `torch.distributed.all_gather_into_tensor` (`distributed/parallel_state.py:2596-2608`, `1246-1248`). The reported `_ALLGATHER_BASE` (`NumelIn=38144, NumelOut=152576` = 1 row × vocab 152576 / attn_tp 4) is the last-token logits gather of the audio request's prefill (`layers/logits_processor.py:926`) — enqueued behind the wedged all-reduce, never executed.

Two further consequences worth noting:

1. **Even when every DP group happens to have audio items, the numerics are wrong**: the full-TP reduce sums partials from unrelated DP groups' audio batches.
2. Video is immune precisely because the ViT honors `--mm-enable-dp-encoder` (`models/mimo_vl.py:261` → `use_data_parallel=True`, no collectives); the audio encoder is not wired to either flag.

Every other multimodal encoder in the tree already passes one of the two flags (`qwen3_vl.py:161`, `kimi_k25.py:130`, `minimax_vl_common.py:108`, `mimo_vl.py:300`); the audio encoder is the one that was missed. `mimo_v2_asr.py` inherits the same defect through `AudioEncoderMixin`.

Note `sglang`'s own `--watchdog-timeout` does not help — the 600 s timeout is `ProcessGroupNCCL`'s collective timeout, not the scheduler watchdog.

### Reproduction

```bash
python -m sglang.launch_server --model-path XiaomiMiMo/MiMo-V2.5 --trust-remote-code \
  --tp 8 --dp 2 --enable-dp-attention --enable-dp-lm-head --mm-enable-dp-encoder \
  --quantization fp8 --attention-backend fa3 --mm-attention-backend fa3
# then send any /v1/chat/completions request whose content includes an audio item
# (audio_only, or video with an audio track). With dp 2, the request lands in one
# DP group and the hang is immediate and deterministic in our runs.
```

### Environment

8×H200 (141 GB), sglang 0.5.18 (also verified the defective code path is unchanged on current `main` @ 97781eb7f), torch 2.13.0+cu130, CUDA 13.0, fa3 attention backend.
